java-checks/src/main/java/org/sonar/java/checks/MagicNumberCheck.java

```diff
@@ -100,9 +118,9 @@ public void visitAnnotation(AnnotationTree annotationTree) {
 
   @Override
   public void visitVariable(VariableTree tree) {
-    // skip static final variables
-    ModifiersTree modifiers = tree.modifiers();
-    if (!(ModifiersUtils.hasModifier(modifiers, Modifier.STATIC) && ModifiersUtils.hasModifier(modifiers, Modifier.FINAL))) {
+    ExpressionTree initializer = tree.initializer();
+    boolean newArray = initializer != null && initializer.is(Kind.NEW_ARRAY) && ((NewArrayTree) initializer).newKeyword() != null;
```

Comment:
Instead of looking for the presence of the `new` keyword, you should check for presence of initializers.
Currently, you will have an issue with the following code :
```
final long[] array = new long[] {3L, 4L}; // Should be compliant
```